client/src/helpers/storeSearch.test.ts

```diff
@@ -0,0 +1,169 @@
+import { preprocess, getTopKStoreIds } from "./storeSearch";
+import { emptyStore } from "src/interfaces";
+
+describe("Test all exported store search functions", () => {
+  describe("Test preprocess", () => {
+    it("preprocesses an empty string", () => {
+      const s = "";
+      const ps = preprocess(s);
+      expect(ps).toEqual("");
+    });
+
+    it("normalizes case correctly", () => {
+      const s = "UPPERCASE lowercase";
+      const ps = preprocess(s);
+      expect(ps).toEqual("Uppercase Lowercase");
+    });
+
+    it("replaces '&' with \"and\"", () => {
+      const s = "A & B";
+      const ps = preprocess(s);
+      expect(ps).toEqual("A And B");
+    });
+
+    it("replaces hyphens with a space", () => {
+      const s = "A-B";
+      const ps = preprocess(s);
+      expect(ps).toEqual("A B");
+    });
+
+    it("removes other non-word characters", () => {
+      const s = 'A().|[]{}`"b';
+      const ps = preprocess(s);
+      expect(ps).toEqual("Ab");
+    });
+
+    it("strips whitespace and replaces multiple spaces with a single space", () => {
+      const s = "   Many   Spaces   ";
+      const ps = preprocess(s);
+      expect(ps).toEqual("Many Spaces");
+    });
+  });
+
+  describe("Test getTopKStoreIds", () => {
+    const testStores = [
+      Object.assign(emptyStore(), { preprocessedName: "Store A" }),
+      Object.assign(emptyStore(), { preprocessedName: "Store Ab" }),
+      Object.assign(emptyStore(), { preprocessedName: "Store Abc" }),
+      Object.assign(emptyStore(), { preprocessedName: "Mart A" }),
+      Object.assign(emptyStore(), { preprocessedName: "Mart Abc" }),
+    ];
+    const threshold = 0;
+
+    it("returns an exact match", () => {
+      const expectedStoreId = 0;
+      const preprocessedQuery = testStores[expectedStoreId].preprocessedName;
+      const storeIds = getTopKStoreIds(
+        testStores,
+        preprocessedQuery,
+        testStores.length,
+        0
+      );
+      expect(storeIds).toContainEqual({
+        dist: { innerEd: 0, leading: 0, trailing: 0 },
+        storeId: expectedStoreId,
+      });
+    });
+
+    it("returns stores containing the query as a substring", () => {
+      const query = "Store";
+      const preprocessedQuery = preprocess(query);
+      const storeIds = getTopKStoreIds(
+        testStores,
+        preprocessedQuery,
+        testStores.length,
+        0
+      );
+      const expectedStoreIds = testStores.filter((store) =>
+        store.preprocessedName.includes(preprocessedQuery)
```

Comment:
IMO instead of test depending on output of a function that is under test, you can just manually create a list of expected stores.

```suggestion
        store.preprocessedName.includes(preprocessedQuery)
```